README.md

```diff
@@ -1,33 +1,230 @@
 # swap-swear-and-swindle-suite
 
-This repository include a basic example of a courtroom contracts game.
+This repository include a basic abstract example of a courtroom contracts game.
 
-Contracts and tests are under contract directory.
-Though I tries to keep the contracts generic as possible ,the "reflector" game as describe below is currently part of the courtroom contract (courtroom.sol). On later phase it will be changes.
 
-## reflector game
-A very simple basic game and see how its fits to the defined swindle game and test various scenarios
-We can do very simple game as followed
+## Courtroom structure
+The courtroom suite includes :
+ - an abstract generic contract "SwearGame" which in a sense conduct the trial: iterate over the specific game witnesses ,trial statuses and verdict accordingly .
+ - a specific game rules contract which defined the trial rules such as transitions statuses,grace periods per state..  .(in our example it is "MirrorRules")
+ - A specific game witnesses contracts which in a sense are "expert witnesses" which can validate submitted evidences.  
```

Comment:
Having gone through all of this i am still a little unclear of how the witnesses are meant to be employed and then transitioned between. I think there should be one default type of witness that is always applicable to *any* game of Sw3.